### PR TITLE
Add io.github.noxworld_dev.OpenNox.

### PR DESCRIPTION
**Pull request by dennwc**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by dennwc** [comment]
bot, build io.github.noxworld_dev.OpenNox

File: io.github.noxworld_dev.OpenNox.yml [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib/i386-linux-gnu
+      - mkdir -p /app/lib/debug/lib/i386-linux-gnu
+      - mkdir -p /app/lib/i386-linux-gnu/GL
+      - install -Dm644 ld.so.conf /app/etc/ld.so.conf
+    sources:
+      - type: file
+        dest-filename: ld.so.conf
+        url: data:/app/lib32%0A/app/lib/i386-linux-gnu%0A
```
Comment (barthalion) [inline io.github.noxworld_dev.OpenNox.yml:65]: Use "inline" type instead.

File: io.github.noxworld_dev.OpenNox.yml [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+    download-if: active-gl-driver
+    enable-if: active-gl-driver
+
+finish-args:
+  - --allow=multiarch
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Wayland access
+  - --socket=wayland
```
Comment (barthalion) [inline io.github.noxworld_dev.OpenNox.yml:44]: If it works on Wayland, replace `x11` with `fallback-x11`.

File: io.github.noxworld_dev.OpenNox.yml [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+        commit: c262619501e6c76b208adbf66d604de07ddbddaa
+        dest: ./res
+
+      # Flatpak files
+      - type: file
+        path: io.github.noxworld_dev.OpenNox.desktop
+      - type: file
+        path: io.github.noxworld_dev.OpenNox.HD.desktop
+      - type: file
+        path: io.github.noxworld_dev.OpenNox.metainfo.xml
```
Comment (barthalion) [inline io.github.noxworld_dev.OpenNox.yml:102]: These files should be ideally maintained in your repo. They are not specific to Flatpak.

File: io.github.noxworld_dev.OpenNox.yml (hunk 2 above)
Comment (txtsd) [reply io.github.noxworld_dev.OpenNox.yml:44]: It does work on Wayland!

File: io.github.noxworld_dev.OpenNox.yml (hunk 1 above)
Comment (dennwc) [reply io.github.noxworld_dev.OpenNox.yml:65]: I copied it from [multiarch docs](https://docs.flatpak.org/en/latest/multiarch.html), will PR a fix there as well.

Comment (dennwc) [reply io.github.noxworld_dev.OpenNox.yml:65]: https://github.com/flatpak/flatpak-docs/pull/393

**Comment by dennwc** [comment]
bot, build io.github.noxworld_dev.OpenNox

**Comment by barthalion** [comment]
bot, build io.github.noxworld_dev.OpenNox

**Comment by barthalion** [comment]
Currently, the app silently fails if it fails to find the datadir. That's something that needs to be changed. User should be either asked what's the location of data files, or told where to place it.

**Comment by dennwc** [comment]
bot, build io.github.noxworld_dev.OpenNox

**Comment by dennwc** [comment]
bot, build io.github.noxworld_dev.OpenNox

**Comment by dennwc** [comment]
@barthalion Done, added a detailed error message pointing to the installation guide.

**Comment by dennwc** [comment]
@barthalion friendly ping :) I think everything is ready here.

**Pull request closed without merging**